types/node/index.d.ts

```diff
@@ -2017,7 +2004,7 @@ declare module "url" {
         hash?: string;
     }
 
-    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
+    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): URL;
```

Comment:
You need three interfaces; Url, URL and requestOptions
parse() returns Url
request() accepts URL and requestOptions